nupic/frameworks/opf/metrics.py

```diff
@@ -1459,7 +1424,63 @@ def addInstance(self, groundTruth, prediction, record = None):
 
     return self.aggregateError
 
+###################################################################################
+class MetricMulti(MetricsIface):
+  """Multi metric can combine multiple other (sub)metrics and 
+     weight them to provide combined score."""
 
+  def __init__(self, metricSpec):
+    """MetricMulti constructor using metricSpec is not allowed."""
+    raise ValueError("MetricMulti cannot be constructed from metricSpec string! "
+                     "Use MetricMulti(id,weights,metrics) constructor instead.")
+
+  def __init__(self, weights, metrics, id=None, window=None):
+    """MetricMulti 
+       @param id - unique name for this metric
+       @param weights - [list of floats] used as weights
+       @param (sub)metrics - [list of metrics] 
+       @param id - (opt) name of the metric
+       @param window - (opt) window size for moving average, or None
+    """
+    if (weights is None or not isinstance(weights, list) or 
+                          not len(weights) > 0 or
+                          not isinstance(weights[0], float)):
+      raise ValueError("MetricMulti requires 'weights' parameter as a [list of floats]")
+    self.weights = weights
+
+    if (metrics is None or not isinstance(metrics, list) or
+                          not len(metrics) > 0 or
+                          not isinstance(metrics[0], MetricsIface)):
+      raise ValueError("MetricMulti requires 'metrics' parameter as a [list of Metrics]")
+    self.metrics = metrics
+    if window is not None:
+      self.movingAvg = MovingAverage(windowSize=window)
+    else:
+      self.movingAvg = None
 
 
+  def addInstance(self, groundTruth, prediction, record = None):
+    err = 0.0
+    subResults = [m.addInstance(groundTruth, prediction, record) for m in self.metrics]
+    for i in xrange(len(self.weights)):
+      try:
+        err += subResults[i]*self.weights[i]
+      except(TypeError):
+        self.err = None
```

Comment:
This try/catch is confusing and can introduce bugs that are hard to catch (if `subResults` contains a string instead of `None`, for example, this metric would silently return `None`). Please refactor.